### PR TITLE
[Aptos Framework] Change total supply from u64 to u128 in Coin module

### PR DESCRIPTION
**Pull request by movekevin**
### Description

u64 might not be sufficient for a coin's total supply. One example is we currently mint u64.max TestCoin to the core resources account so we're running into overflow issues if we turn on supply monitoring for TestCoin.

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/Coin.move [hunk 1]
```diff
@@ -264,7 +264,7 @@ module AptosFramework::Coin {
         let supply = &mut borrow_global_mut<CoinInfo<CoinType>>(coin_addr).supply;
         if (Option::is_some(supply)) {
             let supply = Option::borrow_mut(supply);
-            *supply = *supply + amount;
+            *supply = *supply + (amount as u128);
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/Coin.move:267]: we should probably check overflow and return proper error code here

**Review by zekun000 — APPROVED** [review]

**Comment by hariria** [comment]
blasphemy

File: aptos-move/framework/aptos-framework/sources/Coin.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/Coin.move]: Done

**Comment by macalinao** [comment]
What are the performance implications of this?

**Comment by zekun000** [comment]
we don't expect any visible perf implications from this, TestCoin currently doesn't enable total supply tracking. Even if it's enabled, 8 more bytes should have negligible impact on performance. The blocker of enabling it is to deal with parallel gas burning, we have prototype in https://github.com/aptos-labs/aptos-core/pull/1555

**Pull request closed without merging**